packages/features/voting-context.tsx

```diff
@@ -188,10 +196,124 @@ function VotingCtxProvider({ children }: ChildrenProps) {
     // TODO save result to db
     setEndMessage('Eleição Encerrada');
     setIsVoting(false);
+    await generateResult();
+  }
+
+  interface ICountVotesElection {
+    [key: string]: number;
+  }
+
+  async function CountVotes(votes: IVote[]) {
+    const countVotes: ICountVotesElection = {};
+
+    for (const vote of votes) {
+      const currentResult = countVotes[vote.code];
+      if (!currentResult) {
+        countVotes[vote.code] = 1;
+        continue;
+      }
+
+      countVotes[vote.code] = currentResult + 1;
+    }
+
+    return countVotes;
+  }
+
+  async function createObjectPostResult(
+    orderResult: [string, number][],
+    election: AvailableElections,
+  ) {
+    const winnerParty: PartySummary = {
+      name: '',
+      members: [],
+      votes: '0',
+    };
+
+    const loserParty: PartySummary = {
+      name: '',
+      members: [],
+      votes: '0',
+    };
+
+    let positionResult: number = 0;
+    for (let index = 0; index < orderResult.length; index++) {
+      if (
+        orderResult[index][0] === 'nulo' ||
+        orderResult[index][0] === 'branco'
+      ) {
+        continue;
+      }
+
+      const party: Party | undefined = partyList.find((party) => {
+        return party.code === orderResult[index][0].toString();
+      });
+
+      if (!party) {
+        return;
+      }
+
+      const members: Array<string> = [];
+
+      members.push(party.members.candidate.name);
+      members.push(party.members.viceCandidate.name);
+
+      if (positionResult === 0) {
+        winnerParty.name = party.name;
+        winnerParty.members = members;
+        winnerParty.votes = orderResult[index][1].toString();
+      }
+
+      if (positionResult === 1) {
+        loserParty.name = party.name;
+        loserParty.members = members;
+        loserParty.votes = orderResult[index][1].toString();
+      }
+      positionResult = positionResult + 1;
+    }
+
+    const resultElection: ResultElection = {
+      electionName: election?.electionName,
+      winnerParty: winnerParty,
+      looserParty: loserParty,
+    };
+
+    return resultElection;
+  }
+
+  async function generateResult() {
+    const votes = await getVotes(currentElectionId);
+
+    const countVotes: ICountVotesElection = await CountVotes(votes);
+
+    const election: AvailableElections | undefined =
+      availableElections.results?.find((election) => {
+        return election.electionId === currentElectionId;
+      });
+
+    if (!election) {
+      return;
+    }
+
+    const orderResult = Object.entries(countVotes).sort((a, b) => b[1] - a[1]);
+
+    const resultElection: ResultElection | undefined =
+      await createObjectPostResult(orderResult, election);
+
+    if (!resultElection) {
+      return;
+    }
+
+    const configDatabase = await getConfiguration();
+
+    electionsApi.postResultElection(
+      configDatabase.resultsDatabaseId,
+      configDatabase.electionDatabaseId,
+      resultElection,
+    );
```

Comment:
tentei simplificar este processo mas n sei se esta tao mais simples assim kkkkk , se achar que tem algo de proveitoso fica aq minha resoluçao
```
  async function postResult() {
    const votes = await getVotes(currentElectionId);
    const configDatabase = await getConfiguration();

    const votedSet = new Set();
    const voteCounts: any = {};

    
// aqui se popula um objeto contendo como chave o numero do partido(unico por isso o new Set() ) e como valor a quantidade de votos     

    votes.forEach((vote) => {
      const code = vote.code;
      if (!isNaN(parseFloat(code))) {
        if (votedSet.has(code))
          return (voteCounts[code] = voteCounts[code]
            ? voteCounts[code] + 1
            : 2);
        votedSet.add(code);
      }
    });
// > acho que dar para otimizar e fazer sem o `voteSet` mas acabei fazendo assim mesmo (:  
  

// aqui se ordena o objeto baseado no `value` das `keys` do objeto voteCounts, observe que é desestruturado os arrays ultilizando os colchetes ` [ ] ` 

    const [[winnerCode, winnerVotes], [loserCode, loserVotes]] = Object.entries(
      voteCounts,
    ).sort((a: any, b: any) => b[1] - a[1]);


// uma vez com a lista ordenada pode se assumir que os dois primeiros sao ó vencedor/perdedor respectivamente

    const [{ members: winnerMembers, name: winnerName }] = partyList.filter(
      (party) => party.code === winnerCode,
    );
    const [{ members: loserMembers, name: loserName }] = partyList.filter(
      (party) => party.code === loserCode,
    );

    electionsApi.postResultElection(
      configDatabase.resultsDatabaseId,
      configDatabase.electionDatabaseId,
      {
        electionName: currentElectionId,
        winnerParty: {
          name: winnerName,
          members: winnerMembers,
          votes: winnerVotes as string,
        },
        looserParty: {
          name: loserName,
          members: loserMembers,
          votes: loserVotes as string,
        },
      },
    );
  }
  // a key `members` fica com o tipo `partyMembers` , apesar de no postResulElection pedir um `string[ ]` entao é necessario mudar os tipos em outros arquivos tambem
```

Links uteis 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
https://flaviocopes.com/how-to-rename-object-destructuring/

Reply:
@victor-carvalho1996 acho que vale seguirmos o caminho do proposto pelo zenha aqui e seguirmos com o review a partir das alterações.

Reply:
Acho que para simplificar ainda mais podemos separar functions exclusivamente de parse de objetos fora da execução principal

Reply:
Foi necessário fazer algumas adaptações a lógica para poder salvar o resultado. Mas o código ficou mais enxuto .